Sources/TuistKit/Mappers/Graph/StaticXCFrameworkModuleMapGraphMapper.swift

```diff
@@ -0,0 +1,182 @@
+                        }
+                    ),
+                    "HEADER_SEARCH_PATHS": .array(
+                        staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies.flatMap { xcframework -> [String] in
+                            guard let moduleMap = xcframework.path.glob("**/module.modulemap").first
+                            else { return [] }
+                            return [
+                                "\"$(SRCROOT)/\(moduleMap.parentDirectory.relative(to: project.path).pathString)\"",
+                            ]
+                        }
+                    ),
+                ]
+            }
+        }
+
+        return (
+            try propagateSettings(graph: graph, settings: settingsToPropagate),
+            sideEffects,
+            environment
+        )
+    }
+
+    private func moduleMapFlag(
+        for xcframework: GraphDependency.XCFramework,
+        derivedDirectory: AbsolutePath,
+        project: Project
+    ) -> String {
+        let name = xcframework.path.basenameWithoutExt
+        return "-fmodule-map-file=\"$(SRCROOT)/\(derivedDirectory.appending(components: name, "Headers", "module.modulemap").relative(to: project.path).pathString)\""
+    }
+
+    private func derivedDirectory(graph: Graph) -> AbsolutePath {
+        let packageDirectoryPath: AbsolutePath
+        if fileHandler.exists(graph.path.appending(component: Constants.SwiftPackageManager.packageSwiftName)) {
+            packageDirectoryPath = graph.path
+        } else {
+            packageDirectoryPath = graph.path.appending(component: Constants.tuistDirectoryName)
+        }
+
+        return packageDirectoryPath
+            .appending(
+                components: [
+                    Constants.SwiftPackageManager.packageBuildDirectoryName,
+                    Constants.DerivedDirectory.dependenciesDerivedDirectory,
+                ]
+            )
+    }
+
+    private func sideEffects(
+        for staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies: [GraphDependency.XCFramework],
+        derivedDirectory: AbsolutePath
+    ) throws -> [SideEffectDescriptor] {
+        try staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies
+            .flatMap { xcframework -> [SideEffectDescriptor] in
+                guard let moduleMap = xcframework.path.glob("**/module.modulemap").first
+                else { return [] }
+                let name = xcframework.path.basenameWithoutExt
+                let umbrellaHeader = moduleMap.parentDirectory.appending(component: "\(name).h")
+                guard fileHandler.exists(umbrellaHeader)
+                else { return [] }
+                return [
+                    .directory(DirectoryDescriptor(path: derivedDirectory.appending(components: name, "Headers"))),
+                    .file(
+                        FileDescriptor(
+                            path: derivedDirectory.appending(components: name, "Headers", "module.modulemap"),
```

Comment:
This file contains `derivedDirectory.appending(components: name, "Headers", ...)` in a few places. I'd pass that `headersDirectory` around, or have a function to obtain the headers directory from the derived one.